resources/waterheater.rb

```diff
@@ -936,6 +935,52 @@ def self.calc_ef_from_uef(uef, type, fuel_type)
     return nil
   end
 
+  def self.calc_tank_areas(act_vol)
+    pi = Math::PI
+    height = 48.0 # inches
+    diameter = 24.0 * ((act_vol * 0.1337) / (height / 12.0 * pi))**0.5 # inches
+    a_top = pi * (diameter / 12.0)**(2.0) / 4.0 # sqft
+    a_side = pi * (diameter / 12.0) * (height / 12.0) # sqft
+    surface_area = 2.0 * a_top + a_side # sqft
+
+    return surface_area, a_side
+  end
+
+  def self.calc_indirect_tank_ua(act_vol, standby_loss, jacket_r, a_side, runner)
+    # Tank geometry
+    surface_area = calc_tank_areas(act_vol)[0]
+
+    if standby_loss.nil? # Swiched to standby_loss equation fit from AHRI database
+      # calculate independent variable SurfaceArea/vol(physically linear to standby_loss/skin_u under test condition) to fit the linear equation from AHRI database
+      sqft_by_gal = surface_area / act_vol # sqft/gal
+      standby_loss = 2.9721 * sqft_by_gal - 0.4732 # linear equation assuming a constant u, F/hr
+      if standby_loss <= 0
+        runner.registerError("A negative water heater standby loss coefficient (UA) was calculated for indirect water heater, double check its volume to be <829 gal, or add its standby loss input with HPXML/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystemextension/StandbyLoss.")
+      end
+    else
+      if standby_loss <= 0
+        runner.registerError("A negative water heater standby loss is found, double check water heater input: HPXML/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystemextension/StandbyLoss.")
+      end
+    end
+    if standby_loss > 10.0
+      runner.registerWarning("Indirect water heater standby loss is over 10.0 F/hr, double check water heater inputs.")
+    end
+
+    # Test conditions
+    cp = 0.999 # Btu/lb-F
+    rho = 8.216 # lb/gal
+    t_amb = 70.0 # F
+    t_tank_avg = 135.0 # F, Test begins at 137-138F stop at 133F
+
+    # UA calculation
+    q = standby_loss * cp * act_vol * rho # Btu/hr
+    ua = q / (t_tank_avg - t_amb) # Btu/hr-F
+
+    # jacket
+    ua = apply_tank_jacket(jacket_r, nil, nil, ua, a_side)
```

Comment:
How about moving the two standby_loss <= 0 error messages to a check on the final calculated UA that occurs here? Then you could have a simple generic error message like [here](https://github.com/NREL/OpenStudio-HPXML/blob/indirect_standby_loss/resources/waterheater.rb#L1170-L1172).

Reply:
Sure I can. Just tried to be more descriptive for each specific error message. I'm ok to combine them.